cupy/_logic/truth.py

```diff
@@ -255,6 +255,60 @@ def setdiff1d(ar1, ar2, assume_unique=False):
     return ar1[in1d(ar1, ar2, assume_unique=True, invert=True)]
 
 
+def setxor1d(ar1, ar2, assume_unique=False):
+    """Find the set exclusive-or of two arrays.
+
+    Parameters
+    ----------
+    ar1, ar2 : cupy.ndarray
+        Input arrays. They are flattend if they are not already 1-D.
+    assume_unique : bool
+        By default, False, i.e. input arrays are not unique.
+        If True, input arrays are assumed to be unique. This can
+        speed up the calculation.
+
+    Returns
+    -------
+    setxor1d : cupy.ndarray
+        Return the sorted, unique values that are in only one
+        (not both) of the input arrays.
+
+    See Also
+    --------
+    numpy.setxor1d
+
+    """
+    if not assume_unique:
+        ar1 = cupy.unique(ar1)
+        ar2 = cupy.unique(ar2)
+
+    aux = cupy.concatenate((ar1, ar2), axis=None)
+    if aux.size == 0:
+        return aux
+
+    aux.sort()
+
+    setxorkernel = cupy.ElementwiseKernel(
+        'raw T X, int32 len',
+        'bool z',
+        '''
+            if (i == 0) {
+                z = (X[0] != X[1]);
+            }
+            else if (i == len - 1) {
+                z = (X[i] != X[i-1]);
+            }
+            else {
+                z = X[i] != X[i-1] && X[i] != X[i+1];
+            }
```

Comment:
How about simplifying as follows?
```suggestion
        z = (i == 0 || X[i] != X[i-1]) && (i == len - 1 || X[i] != X[i+1])
```